src/inspector_agent.cc

```diff
@@ -191,143 +193,117 @@ static int StartDebugSignalHandler() {
 }
 #endif  // _WIN32
 
-class JsBindingsSessionDelegate : public InspectorSessionDelegate {
+class JSBindingsConnection : public AsyncWrap {
  public:
-  JsBindingsSessionDelegate(Environment* env,
-                            Local<Object> session,
-                            Local<Object> receiver,
-                            Local<Function> callback)
-                            : env_(env),
-                              session_(env->isolate(), session),
-                              receiver_(env->isolate(), receiver),
-                              callback_(env->isolate(), callback) {
-    session_.SetWeak(this, JsBindingsSessionDelegate::Release,
-                     v8::WeakCallbackType::kParameter);
-  }
-
-  ~JsBindingsSessionDelegate() override {
-    session_.Reset();
-    receiver_.Reset();
-    callback_.Reset();
-  }
+  class JSBindingsSessionDelegate : public InspectorSessionDelegate {
+   public:
+    JSBindingsSessionDelegate(Environment* env,
+                              JSBindingsConnection* connection)
+                              : env_(env),
+                                connection_(connection) {
+    }
 
-  bool WaitForFrontendMessageWhilePaused() override {
-    return false;
-  }
+    bool WaitForFrontendMessageWhilePaused() override {
+      return false;
+    }
 
-  void SendMessageToFrontend(const v8_inspector::StringView& message) override {
-    Isolate* isolate = env_->isolate();
-    v8::HandleScope handle_scope(isolate);
-    Context::Scope context_scope(env_->context());
-    MaybeLocal<String> v8string =
-        String::NewFromTwoByte(isolate, message.characters16(),
-                               NewStringType::kNormal, message.length());
-    Local<Value> argument = v8string.ToLocalChecked().As<Value>();
-    Local<Function> callback = callback_.Get(isolate);
-    Local<Object> receiver = receiver_.Get(isolate);
-    callback->Call(env_->context(), receiver, 1, &argument)
-        .FromMaybe(Local<Value>());
-  }
+    void SendMessageToFrontend(const v8_inspector::StringView& message)
+        override {
+      Isolate* isolate = env_->isolate();
+      v8::HandleScope handle_scope(isolate);
```

Comment:
nit: no need for the `v8::`; already `using v8::HandleScope`.

Reply:
Right. This part of the code was just being moved around, but I'll fix it before landing.